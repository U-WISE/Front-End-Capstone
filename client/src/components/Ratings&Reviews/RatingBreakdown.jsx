import React from 'react';

const RatingBreakdown = () => (
  <div>
    <div>RatingBreakdown</div>
  </div>
);
export default RatingBreakdown;
