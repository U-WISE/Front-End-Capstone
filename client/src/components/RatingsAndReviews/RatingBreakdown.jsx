import React, { useState, useEffect, useContext } from 'react';
import { reviewsContext } from '../../contexts/index.js';
import * as utils from './utils/RatingsAndReviews.utils.js';

const RatingBreakdown = () => {
  const { reviews, setFiltered, filtered } = useContext(reviewsContext);
  const reviewData = utils.getReviewData(reviews);
  const filters = utils.makeFiltersLabel(filtered);

  const clearFilters = () => {
    setFiltered({
      'one': false, 'two': false, 'three': false, 'four': false, 'five': false
    });
  };

  const showFilters = () => {
    if (filtered.five
      || filtered.four
      || filtered.three
      || filtered.two
      || filtered.one) {
      return (
        <span id="appliedFilters">Applied Filters: {filters}<button className="linkButtons" onClick={clearFilters}>Clear Filters</button></span>
      );
    }
    return null;
  };

  if (reviews.length === 0) {
    return (
      <div>No reviews found for these filters
        <button className="linkButtons" onClick={clearFilters}> Clear Filters
        </button>
      </div>
    );
  }

  return (
    <div id="RatingsBreakdown">
      <div id="RBStars">Rating Stars</div>
      <div id="RBRating">{reviewData.ratingNumber}</div>
      <h5 id="RatingBreakdown">Rating Breakdown</h5>
      {showFilters()}

      <button
        className="linkButtons"
        id="fiveStar"
        onClick={() => {
          setFiltered({
            'one': filtered.one, 'two': filtered.two, 'three': filtered.three, 'four': filtered.four, 'five': !filtered.five
          });
        }}
      >5 stars
      </button>
      <div className="borderBar" id="fiveStarBar"><div className="BarFill" style={{ width: `${reviewData.fivePercentage}%` }}></div></div>
      <div id="fiveStarCount">{reviewData.fiveStarReviews}</div>

      <button
        className="linkButtons"
        id="fourStar"
        onClick={() => {
          setFiltered({
            'one': filtered.one, 'two': filtered.two, 'three': filtered.three, 'four': !filtered.four, 'five': filtered.five
          });
        }}
      >4 stars
      </button>
      <div className="borderBar" id="fourStarBar"><div className="BarFill" style={{ width: `${reviewData.fourPercentage}%` }}></div></div>
      <div id="fourStarCount">{reviewData.fourStarReviews}</div>

      <button
        className="linkButtons"
        id="threeStar"
        onClick={() => {
          setFiltered({
            'one': filtered.one, 'two': filtered.two, 'three': !filtered.three, 'four': filtered.four, 'five': filtered.five
          });
        }}
      >3 stars
      </button>
      <div className="borderBar" id="threeStarBar"><div className="BarFill" style={{ width: `${reviewData.threePercentage}%` }}></div></div>
      <div id="threeStarCount">{reviewData.threeStarReviews}</div>

      <button
        className="linkButtons"
        id="twoStar"
        onClick={() => {
          setFiltered({
            'one': filtered.one, 'two': !filtered.two, 'three': filtered.three, 'four': filtered.four, 'five': filtered.five
          });
        }}
      >2 stars
      </button>
      <div className="borderBar" id="twoStarBar"><div className="BarFill" style={{ width: `${reviewData.twoPercentage}%` }}></div></div>
      <div id="twoStarCount">{reviewData.twoStarReviews}</div>

      <button
        className="linkButtons"
        id="oneStar"
        onClick={() => {
          setFiltered({
            'one': !filtered.one, 'two': filtered.two, 'three': filtered.three, 'four': filtered.four, 'five': filtered.five
          });
        }}
      >1 star
      </button>
      <div className="borderBar" id="oneStarBar"><div className="BarFill" style={{ width: `${reviewData.onePercentage}%` }}></div></div>
      <div id="oneStarCount">{reviewData.oneStarReviews}</div>

      <div id="PercentReccomend">{`${reviewData.recommended}% of reviews reccomended this product`}</div>
    </div>
  );
};
export default RatingBreakdown;
