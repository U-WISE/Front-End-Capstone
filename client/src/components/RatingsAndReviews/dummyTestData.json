{
  "reviews": [
    {
      "review_id": 720891,
      "rating": 5,
      "summary": "This product was great!",
      "recommend": true,
      "response": "",
      "body": "I really did or did not like this product based on whether it was sustainably sourced. Then I found out that its made from nothing at all.",
      "date": "2019-01-01T00:00:00.000Z",
      "reviewer_name": "funtime",
      "helpfulness": 8,
      "photos": []
    },
    {
      "review_id": 720892,
      "rating": 4,
      "summary": "This product was ok!",
      "recommend": false,
      "response": "",
      "body": "I really did not like this product solely because I am tiny and do not fit into it.",
      "date": "2019-01-11T00:00:00.000Z",
      "reviewer_name": "mymainstreammother",
      "helpfulness": 2,
      "photos": []
    }
  ],
  "reviewsMeta": {
    "product_id": "44388",
    "ratings": {
      "4": "1",
      "5": "1"
    },
    "recommended": {
      "false": "1",
      "true": "1"
    },
    "characteristics": {
      "Fit": {
        "id": 148890,
        "value": "4.0000000000000000"
      },
      "Length": {
        "id": 148891,
        "value": "3.5000000000000000"
      },
      "Comfort": {
        "id": 148892,
        "value": "5.0000000000000000"
      },
      "Quality": {
        "id": 148893,
        "value": "4.0000000000000000"
      }
    }
  },
  "filtered": {
    "one": false,
    "two": false,
    "three": false,
    "four": false,
    "five": false
  }
}