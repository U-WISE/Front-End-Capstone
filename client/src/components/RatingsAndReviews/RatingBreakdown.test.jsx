import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import RatingBreakdown from './RatingBreakdown.jsx';
import { reviewsContext } from '../../contexts/index.js';
import { reviews, filtered } from './dummyTestData.json';

test('to be defined"', () => {
  const component = renderer.create(
    <reviewsContext.Provider value={{ reviews, filtered }}>
      <RatingBreakdown />
    </reviewsContext.Provider>
  );
  expect(component).toBeDefined();
});

test('to undefined', () => {
  const component = renderer.create(
    <reviewsContext.Provider value={{ reviews, filtered }}>
      <RatingBreakdown />
    </reviewsContext.Provider>
  );
  expect(component.anUndefinedValue).toBeUndefined();
});

test('to match snapshot', () => {
  const component = renderer.create(
    <reviewsContext.Provider value={{ reviews, filtered }}>
      <RatingBreakdown />
    </reviewsContext.Provider>
  );
  expect(component).toMatchSnapshot();
});