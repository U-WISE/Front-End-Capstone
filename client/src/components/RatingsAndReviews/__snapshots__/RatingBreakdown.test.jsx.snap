// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to match snapshot 1`] = `
<div
  id="RatingsBreakdown"
>
  <div
    id="RBStars"
  >
    Rating Stars
  </div>
  <div
    id="RBRating"
  >
    4.5
  </div>
  <h5
    id="RatingBreakdown"
  >
    Rating Breakdown
  </h5>
  <span
    id="appliedFilters"
  >
    Applied Filters: 
    Five Stars Four Stars Three Stars Two Stars One Star 
    <button
      className="linkButtons"
      onClick={[Function]}
    >
      Clear Filters
    </button>
  </span>
  <button
    className="linkButtons"
    id="fiveStar"
    onClick={[Function]}
  >
    5 stars
  </button>
  <div
    className="borderBar"
    id="fiveStarBar"
  >
    <div
      className="BarFill"
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
  <div
    id="fiveStarCount"
  >
    1
  </div>
  <button
    className="linkButtons"
    id="fourStar"
    onClick={[Function]}
  >
    4 stars
  </button>
  <div
    className="borderBar"
    id="fourStarBar"
  >
    <div
      className="BarFill"
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
  <div
    id="fourStarCount"
  >
    1
  </div>
  <button
    className="linkButtons"
    id="threeStar"
    onClick={[Function]}
  >
    3 stars
  </button>
  <div
    className="borderBar"
    id="threeStarBar"
  >
    <div
      className="BarFill"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
  <div
    id="threeStarCount"
  >
    0
  </div>
  <button
    className="linkButtons"
    id="twoStar"
    onClick={[Function]}
  >
    2 stars
  </button>
  <div
    className="borderBar"
    id="twoStarBar"
  >
    <div
      className="BarFill"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
  <div
    id="twoStarCount"
  >
    0
  </div>
  <button
    className="linkButtons"
    id="oneStar"
    onClick={[Function]}
  >
    1 star
  </button>
  <div
    className="borderBar"
    id="oneStarBar"
  >
    <div
      className="BarFill"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
  <div
    id="oneStarCount"
  >
    0
  </div>
  <div
    id="PercentReccomend"
  >
    50.0% of reviews reccomended this product
  </div>
</div>
`;
